### PR TITLE
Update evmone version to v0.4 in docs and scripts

### PR DESCRIPTION
**Pull request by a3d4**
Inspired by eec1998.

### CODE REVIEW

**Comment by axic** [comment]
Could also update the `test/evmc/README` to say evmc 7.1.0.

**Comment by bshastry** [comment]
Please note that this update must propagate to the following places, otherwise linux based CIs will continue to use the older `0.4.0`. I am not sure if this is critical but I guess it is good to keep test CIs identical if possible.

- https://github.com/ethereum/solidity/blob/333322ebf4cde86c51653d5736796df63d476d0a/.circleci/docker/Dockerfile.ubuntu1804#L78-L88

- https://github.com/ethereum/solidity/blob/333322ebf4cde86c51653d5736796df63d476d0a/.circleci/docker/Dockerfile.ubuntu1904#L78-L88

- https://github.com/ethereum/solidity/blob/333322ebf4cde86c51653d5736796df63d476d0a/.circleci/docker/Dockerfile.ubuntu1904.clang#L93-L108

- https://github.com/ethereum/solidity/blob/333322ebf4cde86c51653d5736796df63d476d0a/.circleci/docker/Dockerfile.ubuntu1604.clang.ossfuzz#L86-L96

I am not sure this list is exhaustive :-)

**Comment by a3d4** [comment]
@axic @bshastry Done :).

**Comment by leonardoalt** [comment]
I guess someone also needs to build+upload the new docker images

**Comment by Marenz** [comment]
I feel like we should try to have only one place where the version is defined and all the others use it.. but the places are so different that it might prove difficult. Maybe a script that does the updating?

**Comment by bshastry** [comment]
> I feel like we should try to have only one place where the version is defined and all the others use it.. but the places are so different that it might prove difficult. Maybe a script that does the updating?

I agree. I wonder what alternatives to script exist: perhaps a single base docker file that exports version strings for dependencies? So, one place to update.

Edit: I think the "ease of maintenance" is a separate PR though.

**Comment by axic** [comment]
Btw the 0.4.1 release only fixes a build issue for Windows, not sure it is worth the hassle upgrading. We were already using 0.4.0.

**Comment by a3d4** [comment]
@bshastry @axic
I reverted changes from 0.4.0 to 0.4.1, but left changes from 0.3.0.
I used 0.4.0 for CI and 0.4.1 for download links and documentation.

**Comment by bshastry** [comment]
Hello. I did not approve this yet because I wasn't sure having a different evmone version for Windows and Linux may cause problems in the future.

I will create a separate issue (created https://github.com/ethereum/solidity/issues/8682) to track this before approving.

**Review by bshastry — APPROVED** [review]
LGTM for now.

File: test/evmc/README.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # EVMC
 
-This is an import of [EVMC](https://github.com/ethereum/evmc) version [7.0.0](https://github.com/ethereum/evmc/releases/tag/v7.0.0).
+This is an import of [EVMC](https://github.com/ethereum/evmc) version [7.1.0](https://github.com/ethereum/evmc/releases/tag/v7.1.0).
```
Comment (chriseth) [inline test/evmc/README.md:3]: No files changed in this directory.

File: .circleci/osx_install_dependencies.sh [hunk 2]
```diff
@@ -52,8 +52,8 @@ then
   rm -rf z3-4.8.7-x64-osx-10.14.6
 
   # evmone
-  wget https://github.com/ethereum/evmone/releases/download/v0.3.0/evmone-0.3.0-darwin-x86_64.tar.gz
-  tar xzpf evmone-0.3.0-darwin-x86_64.tar.gz -C /usr/local
-  rm -f evmone-0.3.0-darwin-x86_64.tar.gz
+  wget https://github.com/ethereum/evmone/releases/download/v0.4.0/evmone-0.4.0-darwin-x86_64.tar.gz
```
Comment (chriseth) [inline .circleci/osx_install_dependencies.sh:55]: Why does this have 0.4.0 instead of 0.4.1?

File: test/evmc/README.md (hunk 1 above)
Comment (a3d4) [reply test/evmc/README.md:3]: See the request https://github.com/ethereum/solidity/pull/8651#issuecomment-612004544

File: .circleci/osx_install_dependencies.sh (hunk 2 above)
Comment (a3d4) [reply .circleci/osx_install_dependencies.sh:55]: Set back to 0.4.0 due to https://github.com/ethereum/solidity/pull/8651#issuecomment-613322157
The idea now is to use 0.4.1 for Windows only.

**Review by chriseth — APPROVED** [review]

**Pull request merged**